cluster/plan.go

```diff
@@ -833,9 +834,26 @@ func (c *Cluster) BuildSchedulerProcess(host *hosts.Host, serviceOptions v3.Kube
 
 	Binds = append(Binds, c.Services.Scheduler.ExtraBinds...)
 
-	healthCheck := v3.HealthCheck{
-		URL: services.GetHealthCheckURL(false, services.SchedulerPort),
+	maxK8s122Version, err := util.StrToSemVer(MaxK8s122Version)
+	if err != nil {
+		logrus.Warn(err)
+	}
+
+	k8sTag, err := util.GetImageTagFromImage(c.SystemImages.Kubernetes)
+	if err != nil {
+		logrus.Warn(err)
 	}
+	k8sSemVer, err := util.StrToSemVer(k8sTag)
+	if err != nil {
+		logrus.Warn(err)
+	}
+	healthCheck := v3.HealthCheck{}
+	if k8sSemVer.LessThan(*maxK8s122Version) {
```

Comment:
Since we are more-or-less ignoring the error from `util.StrToSemVer`, do we want to add a `nil`-check for `k8sSemVer` here as well to avoid a possible nil-pointer dereference?